tests/test_merge_db_functions.py

```diff
@@ -0,0 +1,337 @@
+# SOFTWARE.
+#
+###############################################################################
+import sys
+import os
+import sqlite3
+
+from tuna.utils.merge_db import parse_jobline
+from tuna.utils.merge_db import parse_text_fdb_name
+from tuna.utils.merge_db import parse_text_pdb_name
+from tuna.utils.merge_db import load_master_list
+from tuna.utils.merge_db import best_solver
+from tuna.utils.merge_db import target_merge
+from tuna.utils.merge_db import no_job_merge
+from tuna.utils.merge_db import single_job_merge
+from tuna.utils.merge_db import multi_job_merge
+from tuna.utils.merge_db import update_master_list
+from tuna.utils.merge_db import write_merge_results
+from tuna.utils.merge_db import merge_text_file
+from tuna.utils.merge_db import merge_sqlite_pdb
+from tuna.utils.merge_db import merge_sqlite_bin_cache
+from tuna.utils.merge_db import merge_sqlite
+from tuna.utils.merge_db import get_file_list
+from tuna.utils.merge_db import get_sqlite_table
+from tuna.utils.merge_db import get_sqlite_row
+from tuna.helper import prune_cfg_dims
+from tuna.utils.merge_db import get_sqlite_data
+
+this_path = os.path.dirname(__file__)
+
+
+def test_merge_db_functions():
+  test_parse_jobline
+  test_parse_text_fdb_name
+  test_parse_text_pdb_name
+  test_load_master_list
+  test_best_solver
+  test_target_merge
+  test_update_master_list
+  test_write_merge_results
+  test_merge_text_file
+  test_get_sqlite_table
+  test_get_sqlite_row
+  test_get_sqlite_data
+  test_get_file_list
+
+
+def test_parse_jobline():
+  data = """1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F=miopenConvolutionFwdAlgoImplicitGEMM:
+        ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,
+        0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used;
+        miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxSf2x3g1,0.03856,0,miopenConvolutionFwdAlgoWinograd,
+        not used;miopenConvolutionFwdAlgoDirect:ConvOclDirectFwdGen,0.0536,0,miopenConvolutionFwdAlgoDirect,not used;
+        miopenConvolutionFwdAlgoGEMM:GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used"""
+
+  key, vals = parse_jobline(data)
+  assert key, vals
+
+
+def test_parse_text_fdb_name():
+  master_file = '/data/tests/gfx90a_68.HIP.fdb.txt'
+
+  arch, num_cu, final_file, copy_files = parse_text_fdb_name(master_file)
+  assert (arch == "gfx90a")
+  assert (num_cu == '68')
+  assert (final_file == "/data/tests/gfx90a_68.HIP.fdb.txt")
+  assert (copy_files == [])
+
+
+def test_parse_text_pdb_name():
+  master_file = '/data/test/gfx1030_36.cd.pdb.txt'
+
+  arch, num_cu, final_file, copy_files = parse_text_pdb_name(master_file)
+  assert (arch == "gfx1030")
+  assert (num_cu == 36)
+  assert (final_file == "/data/tests/gfx1030_36.cd.pdb.txt")
+  assert (copy_files == [])
+
+
+def test_load_master_list():
+  master_file = '/data/tests/gfx803_36.HIP.fdb.txt'
+  master_list = load_master_list(master_file)
+  if master_list == {}:
```

Comment:
there should be some values in master_list correct? It should check for something other than empty or not

Reply:
added checks for the content of master list.